metaflow/plugins/kfp/kfp_cli.py

```diff
@@ -62,164 +61,169 @@ def step_init(obj, run_id, step_name, passed_in_split_indexes, task_id):
-    @click.option(
-        "--sqs-url-on-error",
-        "-su",
-        "sqs_url_on_error",
-        default=from_conf("METAFLOW_SQS_URL_ON_ERROR", default=None),
-        help="SQS url to send messages upon error"
-        "If not set, messages will NOT be sent to SQS",
-        show_default=True,
-    )
-    @click.option(
-        "--sqs-role-arn-on-error",
-        "-sra",
-        "sqs_role_arn_on_error",
-        default=from_conf("METAFLOW_SQS_ROLE_ARN_ON_ERROR", default=None),
-        help="aws iam role used for sending messages to SQS upon error"
-        "If not set, the default iam role associated with the pod will be used",
-        show_default=True,
-    )
-    @functools.wraps(func)
-    def wrapper_common_options(*args, **kwargs):
-        return func(*args, **kwargs)
+def common_create_run_options(run_type):
+    def cli_decorator(func: Callable):
+        @click.option(
+            "--name",
+            "--pipeline-name",
+            "name",
+            default=None,
+            help="The workflow name. The default is the flow name.",
+            show_default=True,
+        )
+        @click.option(
+            "--experiment",
+            "-e",
+            "experiment",
+            default=None,
+            help="The associated experiment name for the run. ",
+            show_default=True,
+        )
+        @click.option(
+            "--tag",
+            "tags",
+            multiple=True,
+            default=None,
+            help="Annotate all objects produced by Argo Workflows runs "
+            "with the given tag. You can specify this option multiple "
+            "times to attach multiple tags.",
+        )
+        @click.option(
+            "--sys-tag",
+            "sys_tags",
+            multiple=True,
+            default=None,
+            help="Annotate all Metaflow objects produced by Argo Metaflow runs "
+            "with the given system tag. You can specify this option multiple "
+            "times to attach multiple tags.",
+        )
+        @click.option(
+            "--namespace",
+            "user_namespace",
+            default=None,
+            help="Change the namespace from the default (production token) "
+            "to the given tag. See run --help for more information.",
+        )
+        @click.option(
+            "--k8s-namespace",
+            "--kubernetes-namespace",
+            "kubernetes_namespace",
+            default=KUBERNETES_NAMESPACE,
+            help="Kubernetes Namespace for your run in Argo.",
+            show_default=True,
+        )
+        @click.option(
+            "--yaml-only",
+            "yaml_only",
+            is_flag=True,
+            default=False,
+            help="Generate the Workflow YAML which is used to run the workflow on Argo.",
+            show_default=True,
+        )
+        @click.option(
+            "--kind",
+            "kind",
+            default="Workflow" if run_type == "run" else "WorkflowTemplate",
```

Comment:
[minor]
It looks like `run_type` is only used here. Instead of having `run_type` be `run` or `create`, can we directly assign `"Workflow"` or `"WorkflowTemplate"` to it? 

This line will then be `default = run_type`.
The two functions calling `common_create_run_options` will be 
```
@common_create_run_options(run_type="Workflow")
def run(...):

@common_create_run_options(run_type="WorkflowTemplate")
def create(...):
```

Reply:
Your current code may be better if we have more of these specialization between `create` and `run`

Reply:
updated to usage example 
```python
@common_create_run_options(default_yaml_kind="WorkflowTemplate")
```